src/workerd/util/sqlite.c++

```diff
@@ -9,28 +9,29 @@
 #include <sys/stat.h>
 #include <sqlite3.h>
 #include <kj/vector.h>
+#include <workerd/jsg/jsg.h>
 
 namespace workerd {
 
 #define SQLITE_CALL_NODB(code, ...) do { \
     int _ec = code; \
-    KJ_ASSERT(_ec == SQLITE_OK, "SQLite failed: " #code, sqlite3_errstr(_ec), ##__VA_ARGS__); \
+    JSG_ASSERT(_ec == SQLITE_OK, Error, "SQLite failed: " #code, " ", sqlite3_errstr(_ec), ##__VA_ARGS__); \
   } while (false)
 // Make a SQLite call and check the returned error code. Use this version when the call is not
 // associated with an open DB connection.
 
 #define SQLITE_CALL(code, ...) do { \
     int _ec = code; \
     /* SQLITE_MISUSE doesn't put error info on the database object, so check it separately */ \
-    KJ_ASSERT(_ec != SQLITE_MISUSE, "SQLite misused: " #code, ##__VA_ARGS__); \
-    KJ_ASSERT(_ec == SQLITE_OK, "SQLite failed: " #code, sqlite3_errmsg(db), ##__VA_ARGS__); \
+    JSG_ASSERT(_ec != SQLITE_MISUSE, Error, "SQLite misused: " #code, ##__VA_ARGS__); \
+    JSG_ASSERT(_ec == SQLITE_OK, Error, "SQLite failed: " #code, " ", sqlite3_errmsg(db), ##__VA_ARGS__); \
```

Comment:
I'm not sure if it's quite safe to just mark all SQLite errors as application errors (using JSG_ASSERT). We might have to define a way for the caller to specify whether a query is coming from the application vs. from "the system".

For example, consider the actor-sqlite code that implements the old DO KV API in terms of sqlite. If that code does something wrong triggering a sqlite error, we want that error to be logged to Sentry; we don't want to pass it through to the application. JSG_ASSERT failures will not go to Sentry since we assume they are the application's fault.

I wonder if this could be combined with the separate problem of application code needing to be prohibited from running certain queries, such as dangerous pragmas, or accessing tables whose names start with `_cf_`. The same flag that distinguishes app vs. system could maybe be used for both purposes.

Reply:
> I'm not sure if it's quite safe to just mark all SQLite errors as application errors...

100% agree. We need to consider these errors carefully.